pkg/tfvars/internal/cache/cache.go

```diff
@@ -96,6 +96,21 @@ func cacheFile(reader io.Reader, filePath string, sha256Checksum string) (err er
 		}
 	}()
 
+	// Make sure that the temp file does not exist after completion of the function
+	defer func() {
+		_, err2 := os.Stat(tempPath)
```

Comment:
Why bother with a racy `Stat`?  Just try `os.Remove(tempPath)` and see if it works.